Client/Helpers/FxALoginHelper.swift

```diff
@@ -0,0 +1,191 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Account
+import Deferred
+import Foundation
+import Shared
+
+private let applicationDidRequestUserNotificationPermissionPrefKey = "applicationDidRequestUserNotificationPermissionPrefKey"
+
+/// This class manages the from successful login for FxAccounts to 
+/// asking the user for notification permissions, registering for 
+/// remote push notifications (APNS), registering for WebPush notifcations
+/// then creating an account and storing it in the profile.
+class FxALoginHelper {
+    private let deferred = Success()
+
+    private let pushClient: PushClient
+
+    private weak var application: UIApplication?
+    private weak var profile: Profile?
+
+    private var account: FirefoxAccount!
+    private var pushRegistration: PushRegistration!
+
+    static private(set) var sharedInstance: FxALoginHelper?
```

Comment:
I've updated the helper to much less stateful: it doesn't store an `UIApplication`, and uses the one used when passed in from the `AppDelegate` and `FxAContentViewController`.

The helper is created when the `Profile` is initially loaded from disk, which corresponds to the time when the app might need to register for push notifications.

This has the effect of: 

 * we can get rid of the `createSharedInstance` methods, and optionality.
 * we can get rid of the `application` optionality.
 * only one assert needs to happen, and that case is guarded against.